File: test/ch21-benchmarks/bench-011-nested-loops.py
#!/usr/bin/env python3
# BENCH-011: Nested Loops (1000x1000 iterations) - Python

def nested_loops(outer, inner):
    total = 0
    for i in range(outer):
        for j in range(inner):
            total += i * j
    return total

result = nested_loops(1000, 1000)
# Expected: 249500250000

File: test/ch21-benchmarks/bench-011-nested-loops.go
// BENCH-011: Nested Loops (1000x1000 iterations) - Go
package main

func nestedLoops(outer, inner int) int {
	total := 0
	for i := 0; i < outer; i++ {
		for j := 0; j < inner; j++ {
			total += i * j
		}
	}
	return total
}

func main() {
	result := nestedLoops(1000, 1000)
	_ = result // Expected: 249500250000
}
